src/main/java/com/example/planit/engine/CalendarEngine.java

```diff
@@ -1398,16 +1397,24 @@ private StudySession getUpComingSessionForUser(User user) throws GeneralSecurity
     }
 
     /**
-     * add break in the end of each user event in the list
+     * add break in the start and end of each user event in the list
      *
      * @param userEvents    list of all the {@link Event} the user have in the calendars
      * @param userBreakTime the break time in user preferences in minutes
      */
-    private static void addBreakTimeInEndOfEachUserEvent(List<Event> userEvents, int userBreakTime) {
+    private static void addBreakTimeInStartAndEndOfEachUserEvent(List<Event> userEvents, int userBreakTime) {
 
         for (Event currentUserEvent : userEvents) {
-            Instant oldEndOfUserEvent = Instant.ofEpochMilli(currentUserEvent.getEnd().getDateTime().getValue());
-            Instant newEndOfUserEvent = oldEndOfUserEvent.plus(Math.min(2 * userBreakTime, SPACE_LIMIT_BETWEEN_EVENTS), ChronoUnit.MINUTES);
+            // add break in the end of the current event
+            Instant oldEndTimeOfUserEvent = Instant.ofEpochMilli(currentUserEvent.getEnd().getDateTime().getValue());
+            Instant newEndOfUserEvent = oldEndTimeOfUserEvent.plus(Math.min(2 * userBreakTime, SPACE_LIMIT_BETWEEN_EVENTS), ChronoUnit.MINUTES);
+            // add break in the start of the current event
```

Comment:
Rename to newEndTimeOfUserEvent please

Reply:
Or remove the "time" from `oldEndTimeOfUserEvent`. So it will be consistant.
I prefer the option including the "time" in the variables' names